src/V3Os.cpp

```diff
@@ -363,3 +372,229 @@ int V3Os::system(const string& command) {
         return exit_code;
     }
 }
+
+//######################################################################
+// METHODS (segfault handler)
+
+#if defined(__linux__)
+class ProcMapsParser final {
+
+    int m_fd = -1;  // file descriptor
+    std::array<char, 256> m_buf = {};  // chunk of data read from the file.
+    const char* m_ccp = nullptr;  // pointer to current character.
+    const char* m_endcp = nullptr;  // pointer to a character one past the end of data in m_buf.
+
+private:
+    // Moves m_ccp to the next character. Reads next chunk of data if needed.
+    void advance() {
+        if (m_ccp != m_endcp) {
+            ++m_ccp;
+        } else {
+            const ssize_t len = read(m_fd, m_buf.data(), m_buf.size());
+            const ssize_t buf_size = static_cast<ssize_t>(m_buf.size());
+            if (len == buf_size) {
+                m_ccp = &m_buf.front();
+                m_endcp = m_ccp + (m_buf.size() - 1);
+            } else if (len > 0 && len < buf_size) {
+                m_buf[len] = '\0';
+                m_ccp = m_buf.data();
+                m_endcp = &m_buf[len];
+            } else {
+                m_buf[0] = '\0';
+                m_ccp = m_buf.data();
+                m_endcp = m_ccp;
+            }
+        }
+    }
+
+    // Moves m_cc to point at the first byte of the next line.
+    void skip_line() {
+        while (*m_ccp != '\n' && *m_ccp != '\0') { advance(); }
+        if (*m_ccp == '\n') { advance(); }
+    }
+
+    uintptr_t parseHexNumber() {
```

Comment:
Use strtol?

Reply:
Not allowed in signal handlers ([`man signal-safety`](https://man7.org/linux/man-pages/man7/signal-safety.7.html)).